### PR TITLE
Character creation fix

### PR DESCRIPTION
**Pull request by Kai7788**
- A few monsters were moved from the data/json/ directory into test files
- The problem is that some of the dps calculation logic is hardcoded to depend on them

- This bug was introduced with #88687
#### Summary
Bugfixes "Can create Characters again"

#### Purpose of change
The character creation screen had a bug where the border was just shown in red and the ui was not drawn correctly.
This prevents everyone from creating characters.
See:
beforeStartedWork

#### Describe the solution

After some attaching a debugger and following stack traces i found the following:
Stack Trace of Missing End()

Invalid Monster type

It crashed because it did not find a specific monster type id.
Looking at the stack trace it seemed to occur in the `item_gun_tool_ammo.cpp` file and here i found a part of the answer.
The specified monster it didn't find is used for some damage calculations (i did not look into which exactly) and they ARE hardcoded.

Looking at some recent Merge Requests i saw #88687 which moved the files for these monsters to the test json files.
This means the needed monsters were not registered and thus broke the UI ( This should be addressed in a separate issue ).
testmonsterReferencedInProductionCode

So the fix was easy. I moved the missing json back and the game loaded as usual.

AfterRestoringTheFiles

#### Describe alternatives you've considered
I do not know this code good enough to make a judgement on what would be right to do here.

#### Testing
After re-adding the json i ran all tests locally and none failed.

#### Additional context

### CODE REVIEW

File: data/json/monsters/zed_soldiers.json [hunk 1]
```diff
@@ -57,6 +57,18 @@
     },
     "delete": { "flags": [ "STUMBLES" ] }
   },
+  {
+    "id": "mon_zombie_soldier_no_weakpoints",
+    "type": "MONSTER",
+    "name": { "str_sp": "zombie soldier test only", "//~": "NO_I18N" },
+    "description": { "str": "This zombie soldier is for testing purposes only.", "//~": "NO_I18N" },
```
Comment (Procyonae) [inline data/json/monsters/zed_soldiers.json]:
```suggestion
    "description": { "str": "This zombie soldier is used by hardcoded dps logic and for testing purposes only.", "//~": "NO_I18N" },
```

File: data/json/monsters/zed_survivor.json [hunk 2]
```diff
@@ -16,6 +16,20 @@
     "extend": { "weakpoint_sets": [ "wps_humanoid_light_armor", "wps_humanoid_open_helmet" ], "families": [ "prof_wp_syn_armored" ] },
     "delete": { "flag": [ "STUMBLES" ] }
   },
+  {
+    "id": "mon_zombie_survivor_no_weakpoints",
+    "type": "MONSTER",
+    "name": { "str_sp": "zombie survivor test only", "//~": "NO_I18N" },
+    "description": { "str": "This zombie survivor is for testing purposes only.", "//~": "NO_I18N" },
```
Comment (Procyonae) [inline data/json/monsters/zed_survivor.json]:
```suggestion
    "description": { "str": "This zombie survivor is used by hardcoded dps logic and for testing purposes only.", "//~": "NO_I18N" },
```

**Review by Procyonae — CHANGES_REQUESTED** [review]
Should probably try to make this less likely to happen again too c;

File: data/json/monsters/zed_soldiers.json (hunk 1 above)
Comment (Kai7788) [reply data/json/monsters/zed_soldiers.json]: Good idea!

**Comment by GuardianDll** [comment]
i wanted to say "it is surprising tests didn't fail with it", but then realised tests run with TEST_DATA mod on, so obviously they could not catch it
